pkg/tfbridge/metadata.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfbridge
+
+import (
+	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/metadata"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+// A store persisted between `tfgen` and a running provider.
+type ProviderMetadata *metadata.Data
+
+type MetadataInfo struct {
+	Path string
```

Comment:
Maybe give a default value to path, like "bridge-metadata.json" used in TFPF was something I came up with.